### PR TITLE
chore(2640): add snapshot tests for check command

### LINKED ISSUE #2640: Add snapshot tests for CLI's check command
**Technical Requirements**
- Add the tests in `tests/cli/check.rs`.
- use `include_config!` macro to import configurations from a `test/fixtures` directory where test configurations should be added.
- Use the CLI programatically and create snapshot tests.
- Take inspiration from the [generator.rs](https://github.com/tailcallhq/tailcall/blob/main/src/cli/generator/generator.rs)

### PR DESCRIPTION
**Pull request by beelchester**
fixes #2640
/claim #2640

### CODE REVIEW

**Comment by ssddOnTop** [comment]
I am working on #2644 .. maybe we can collab on this one

**Comment by beelchester** [comment]
Sure 👍

File: src/cli/tc/check.rs [hunk 1]
```diff
@@ -8,28 +10,45 @@ use crate::core::config::reader::ConfigReader;
...
+    pub schema: bool,
+    pub format: Option<Source>,
+    pub runtime: TargetRuntime,
 }
 
-pub(super) async fn check_command(params: CheckParams, config_reader: &ConfigReader) -> Result<()> {
+pub async fn check_command(
+    params: CheckParams,
+    config_reader: &ConfigReader,
+    mut write_buf: Option<&mut dyn Write>,
```
Comment (tusharmath) [inline src/cli/tc/check.rs:24]: I think we can hook in a custom tracing module in the tests. We can then assert what was written to the buffer.

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Pull request closed without merging**